Private vendor work (#81)
anima-wg/voucher: draft-ietf-anima-rfc8366bis.md

Section 5.1 of {{RFC5652}}, encoded using ASN.1 Distinguished Encoding
Rules (DER), as specified in ITU-T X.690 {{ITU-T.X690.2015}}.

> To facilitate interoperability, {{vcj}} the media type "application/voucher-cms+json" and the filename extension ".vcj". were registered by {{RFC8366}}.

Reviewer: ```suggestion
To facilitate interoperability, {{vcj}} the media type "application/voucher-cms+json" and the filename extension ".vcj" were registered by {{RFC8366}}.
```

The JSON serialization of vouchers implicitely accomodates the above, since the voucher is just a map (or dictionary).
Map keys are just strings, and creating unique strings is easy to do by including the manufacturer's domain name.

In CBOR serialization {{RFC9148}}/{{RFC9254}}, the situation is not so easy.
The delta encoding for keys requires new keys to use the absolute Tag(47) for new entities.
An extension might need to use the Private Use SID values, or acquire SID values for their own use.

Where the process has become complex is when making standard extensions, as has happened recently, resulting in this document.
The processes which were anticipated to be useful, (the "augment" mechanism) turned out not to be the case, see {{extendfail}}.

Instead, a process similiar to what was done by {{?RFC8520}} has been adopted.

> In this, extensions are listed in a new leaf by that name.

Reviewer: "by that name" - does this mean the name of the leaf is literally "extensions" ? If so, best to make that explicit. E.g.

   extensions are listed in a leaf named "extensions".

Reviewer: was resolved in a later commit.

> In this, extensions are listed in a new leaf by that name.

Author: ```suggestion
In this, extensions are listed in a leaf named "extensions"
```

> In JSON serialization, these extensions require a unique name, and this MUST be allocated with IANA in the

Reviewer: sentence partly missing?

Reviewer: This was resolved in a later commit.

> In JSON serialization, these extensions require a unique name, and this MUST be allocated with IANA in the

Author: ```suggestion
In JSON serialization, these extensions require a unique name, and this MUST be allocated by IANA.
The name MUST be the same as the YANG module name.
```

The JSON serialization of vouchers implicitely accomodates the above, since the voucher is just a map (or dictionary).
Map keys are just strings, and creating unique strings is easy to do by including the manufacturer's domain name.

In CBOR serialization {{RFC9148}}/{{RFC9254}}, the situation is not so easy.
The delta encoding for keys requires new keys to use the absolute Tag(47) for new entities.
An extension might need to use the Private Use SID values, or acquire SID values for their own use.

Where the process has become complex is when making standard extensions, as has happened recently, resulting in this document.
The processes which were anticipated to be useful, (the "augment" mechanism) turned out not to be the case, see {{extendfail}}.

Instead, a process similiar to what was done by {{?RFC8520}} has been adopted.

> In this, extensions are listed in a leaf named "extensions"

Reviewer: ```suggestion
In this, extensions are listed in a leaf named "extensions".
```

As the Voucher Request artifact is created by augment on the voucher artifact, any extension defined for the voucher is also valid for Voucher Requests.

## Manufacturer Private extensions

In addition to the above described extensions mechanism, a manufacturer might need to communicate content in the voucher (or in the voucher-request), which are never subject to standardization.

> While they can use the mechanism above, it does require allocation of a SID in order to do minimal sized encoding.  Note that {{RFC9148}} does not require use of SIDs.

Reviewer: another RFC reference was intended here. (EST-coaps doesn't use vouchers)

Reviewer: was resolved in a later commit

> While they can use the mechanism above, it does require allocation of a SID in order to do minimal sized encoding.  Note that {{RFC9148}} does not require use of SIDs.

Author: ```suggestion
While they can use the mechanism above, it does require allocation of a SID in order to do minimal sized encoding.  Note that {{RFC9254}} does not require use of SIDs.
```
